### PR TITLE
Auth bug

### LINKED ISSUE #440: Raw gobbles up auth-protected endpoint?
Using the tutorial, if I have this API:

```haskell
type Api = "users" :> BasicAuth "here" Bool :> Get '[JSON] [User]
```

Everything works as expected when browsing with my browser. It asks for creds.

However,

```haskell
type Api = "users" :> BasicAuth "here" Bool :> Get '[JSON] [User] :<|> Raw
```
(and serving a local directory) makes it impossible for me to visit the "users" endpoint. I get `404 File not found`.

Am I doing something wrong?

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #440.

Please review.

Also, how do I create basic auth tokens? (https://github.com/haskell-servant/servant/blob/auth-bug/servant-server/test/Servant/ServerSpec.hs#L576)

CC: @aaronlevin @3noch @jkarni @kosmikus @alpmestan

### CODE REVIEW

File: servant-server/test/Servant/ServerSpec.hs [hunk 1]
```diff
@@ -30,6 +30,7 @@ import           Network.HTTP.Types         (Status (..), hAccept, hContentType,
                                              methodDelete, methodGet,
                                              methodHead, methodPatch,
                                              methodPost, methodPut, ok200,
+                                             imATeaPot418,
```
Comment (3noch) [inline servant-server/test/Servant/ServerSpec.hs]: How did we ever do without this all this time?

**Comment by 3noch** [comment]
@soenkehahn You've officially made my day!

**Comment by kosmikus** [comment]
I am ok with this change. (Generally, I am still wary of anything that makes use of `FailFatal`, because it is inherently non-modular. But currently, this definitely looks like the right thing to do.)

**Comment by arianvp** [comment]
```
token user password = base64 $ user ++ ":" ++ password
```

**Comment by alpmestan** [comment]
@soenkehahn What Arian just described is also [in servant-client](https://github.com/haskell-servant/servant/blob/master/servant-client/src/Servant/Common/BasicAuth.hs).

Regarding the PR, looks good to me!

**Comment by soenkehahn** [comment]
I'm going to merge once CI passes. Thanks for the reviews.

I suspect that the general auth is having the same problem, btw.

**Comment by aaronlevin** [comment]
@soenkehahn thanks for fixing this so quickly. LGTM.

General auth uses `FailFatal`, so it should be ok.

**Comment by aaronlevin** [comment]
(though we should probably add a test case - I can do that now)

**Comment by aaronlevin** [comment]
@soenkehahn i'm going to push to this branch with a change to test Gen Auth.

**Comment by soenkehahn** [comment]
@aaronlevin: Can you use a separate PR, please?

**Comment by soenkehahn** [comment]
@aaronlevin: Sorry, removed your commit from this PR. Please create a new one.

**Comment by aaronlevin** [comment]
@soenkehahn no worries. submitted.

**Pull request merged**